.tekton/.currency/scripts/generate_report.py

```diff
@@ -0,0 +1,243 @@
+            break
+        else:
+            print(
+                f"Unable to retrieve container logs from the successful taskrun pod {pod_name} of taskrun {taskrun_name}."
+            )
+    return tekton_ci_output, check_net_http
+
+
+def get_tekton_ci_output():
+    # config.load_kube_config()
+    config.load_incluster_config()
+
+    namespace = "default"
+    core_v1_client = client.CoreV1Api()
+
+    default_libraries_dict = {
+        "cuba": 1,
+        "excon": 4,
+        "graphql": 6,
+        "grpc": 7,
+        "rack": 10,
+        "rest-client": 11,
+        "roda": 13,
+        "sinatra": 16,
+    }
+
+    tekton_ci_output = ""
+    task_name = "ruby-tracer-unittest-default-libraries-task"
+    default_taskruns = get_taskruns(namespace, task_name)
+
+    check_net_http=True
+
+    for library, pattern in default_libraries_dict.items():
+        taskrun_filter = (
+            lambda tr: tr["metadata"]["name"].endswith(
+                f"unittest-default-ruby-33-{pattern}"
+            )
+            and tr["status"]["conditions"][0]["type"] == "Succeeded"
+        )
+        filtered_default_taskruns = filter_taskruns(taskrun_filter, default_taskruns)
+
+        tekton_ci_output, check_net_http = process_taskrun_logs(
+            filtered_default_taskruns,
+            core_v1_client,
+            namespace,
+            library,
+            tekton_ci_output,
+            check_net_http=check_net_http
+        )
+
+    other_libraries_dict = {
+        "rails": {
+            "pattern": "rails-postgres-11",
+            "task_name": "ruby-tracer-unittest-rails-postgres-task",
+        },
+        "dalli": {
+            "pattern": "memcached-11",
+            "task_name": "ruby-tracer-unittest-memcached-libraries-task",
+        },
+        "resque": {
+            "pattern": "unittest-redis-ruby-32-33-9",
+            "task_name": "ruby-tracer-unittest-redis-libraries-task",
+        },
+        "sidekiq": {
+            "pattern": "unittest-redis-ruby-32-33-18",
+            "task_name": "ruby-tracer-unittest-redis-libraries-task",
+        },
+    }
+
+    for library, inner_dict in other_libraries_dict.items():
+        pattern = inner_dict["pattern"]
+        task_name = inner_dict["task_name"]
+        taskrun_filter = (
+            lambda tr: tr["metadata"]["name"].endswith(pattern)
+            and tr["status"]["conditions"][0]["type"] == "Succeeded"
+        )
+        other_taskruns = get_taskruns(namespace, task_name)
+        filtered_other_taskruns = filter_taskruns(taskrun_filter, other_taskruns)
+
+        tekton_ci_output, check_net_http = process_taskrun_logs(
+            filtered_other_taskruns,
+            core_v1_client,
+            namespace,
+            library,
+            tekton_ci_output,
+            check_net_http=False
+        )
+
+    return tekton_ci_output
+
+
+def get_latest_stable_ruby_version():
+    url = "https://www.ruby-lang.org/en/downloads/"
```

Comment:
I think GH is an equally valid source of information here, and GH has a stable JSON API, which is much more robust than trying to parse the ruby website.

See https://api.github.com/repos/ruby/ruby/releases
and https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#list-releases

Reply:
That's interesting to know! I will use these for the additional changes requested by the tracer community.

But for now, I will remove this function since this is out of scope for the current PR. I used this function earlier to get the latest version for Net::HTTP std lib.